orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -147,18 +151,27 @@ def __init__(self, model, model_desc, optim_config, loss_fn=None, options=None):
         self.model_desc = _ORTTrainerModelDesc(model_desc)
         self.optim_config = optim_config
 
+        # ORTTrainerOptions
         if not options:
             options = ORTTrainerOptions()
         self.options = options
 
+        # Post processing ONNX model given as input
+        if self._onnx_model:
+            if self.options._internal_use.enable_internal_postprocess:
+                self._onnx_model = postprocess.run_postprocess(self._onnx_model)
+            if self.options._internal_use.extra_postprocess:
+                self._onnx_model = self.options._internal_use.extra_postprocess(self._onnx_model)
+
         # Set GPU device and memory limit
-        device_id = self.options.device.id
-        if 'cuda' in device_id.lower():
-            set_cuda_mem_limit(int(self.options.device.mem_limit))
-            if ':' in device_id:
-                set_cuda_device_id(int(device_id.split(':')[1]))
+        if 'cuda' in self.options.device.id.lower():
+            mem_limit = self.options.device.mem_limit
+            if  mem_limit > 0:
```

Comment:
Should we try and warn the user if they pick something too small? If they pick 256 bytes and it doesn't crash it may be hard to debug the issue

Reply:
I raised a bug to the backend reporting this behavior. Let's see what they can do in this